### PR TITLE
Enable use_labels in Data-Parallel CrossEntropy layer 

### PR DESCRIPTION
**Pull request by szaman19**
This PR enables `use_labels` on the channel (0-th dimension of a sample) for CrossEntropy loss. The expected inputs for the cross entropy layer with `use_labels' enabled are: 
- input: (C, d_0, d_1, ..., d_k)
- labels: (1, d_0, d_1, ..., d_k)

To do: 
- [x] Add cross_entropy with labels CI test
- [x] Add expected tensor dimension checks for `input` and `label`
- [x] Add/augment GPU kernel with `use_label` enabled 
- [x] Add CPU implementation

Local CI passing. Pinging @bvanessen @KIwabuchi

### CODE REVIEW

**Comment by KIwabuchi** [comment]
Hey @szaman19 
When I tried to build this version, I got a build error.
I investigated it quickly. You might have accidentally broken a '{ }' pair in `lbann/layers/loss/cross_entropy_impl.hpp`.
Could you look into that?

Here is the command I used:
```shell
# Tested on Pascal
./scripts/build_lbann.sh -d -l shehtab-ce -p py-pip -- +cuda +numpy  +vision
```

Here is the error message:
```shell
    344    In file included from /dev/shm/lbann/src/layers/loss/cross_entropy.cpp:28:
  >> 345    /dev/shm/lbann/include/lbann/layers/loss/cross_entropy_impl.hpp:158:36: error: 'data_layout' has not been declared
     346      158 | template <typename TensorDataType, data_layout T_layout, El::Device Dev>
     347          |                                    ^~~~~~~~~~~
  >> 348    /dev/shm/lbann/include/lbann/layers/loss/cross_entropy_impl.hpp:159:25: error: expected initializer before '<' token
     349      159 | void cross_entropy_layer<TensorDataType, T_layout, Dev>::setup_data(
     350          |                         ^
  >> 351    /dev/shm/lbann/include/lbann/layers/loss/cross_entropy_impl.hpp:185:36: error: 'data_layout' has not been declared
     352      185 | template <typename TensorDataType, data_layout T_layout, El::Device Dev>
     353          |                                    ^~~~~~~~~~~
  >> 354    /dev/shm/lbann/include/lbann/layers/loss/cross_entropy_impl.hpp:186:25: error: expected initializer before '<' token
     355      186 | void cross_entropy_layer<TensorDataType, T_layout, Dev>::fp_compute()
     356          |                         ^
  >> 357    /dev/shm/lbann/include/lbann/layers/loss/cross_entropy_impl.hpp:209:36: error: 'data_layout' has not been declared
     358      209 | template <typename TensorDataType, data_layout T_layout, El::Device Dev>
     359          |                                    ^~~~~~~~~~~
  >> 360    /dev/shm/lbann/include/lbann/layers/loss/cross_entropy_impl.hpp:210:25: error: expected initializer before '<' token
     361      210 | void cross_entropy_layer<TensorDataType, T_layout, Dev>::bp_compute()
     362          |                         ^
  >> 363    /dev/shm/lbann/include/lbann/layers/loss/cross_entropy_impl.hpp:229:23: error: 'data_layout' has not been declared
     364      229 | template <typename T, data_layout L, El::Device D>
     365          |                       ^~~~~~~~~~~
  >> 366    /dev/shm/lbann/include/lbann/layers/loss/cross_entropy_impl.hpp:230:25: error: expected initializer before '<' token
     367      230 | void cross_entropy_layer<T, L, D>::write_specific_proto(
     368          |                         ^
  >> 369    /dev/shm/lbann/include/lbann/layers/loss/cross_entropy_impl.hpp:411:1: error: expected declaration before '}' token
     370      411 | } // namespace lbann
```

**Review by bvanessen — APPROVED** [review]
